src/tls.c

```diff
@@ -3187,7 +3194,13 @@ static int TLSX_CSR_Parse(WOLFSSL* ssl, const byte* input, word16 length,
                     ret = BUFFER_ERROR;
             }
             if (ret == 0) {
-                csr->response.buffer = (byte*)(input + offset);
+                csr->response.buffer = (byte*)XMALLOC(resp_length, NULL,
```

Comment:
Please add heap hint `ssl->heap`.

Reply:
Added a way to use a heap hint. Also removed `responseBuffer` from internal API's that don't use it anywhere.